### PR TITLE
[RORDEV-1329] ESQL support

### PR DESCRIPTION
**Pull request by coutoPL**
🚨Security Fix (ES) [data leak through the ESQL API](https://forum.readonlyrest.com/t/eql-requests-returns-data-even-though-they-aren-t-allowed/2679) (for ES >= 8.11.0)
🚀New (ES) ESQL support

### CODE REVIEW

File: tests-utils/src/main/scala/tech/beshu/ror/utils/misc/EsModule.scala [hunk 1]
```diff
@@ -64,6 +64,7 @@ trait EsModulePatterns {
   val allEs8x = "^es8\\d+x$".r
   val allEs8xBelowEs87x = "^es8[0-6]x$".r
   val allEs8xAboveEs86x = "^es8([7-9]|[1-9][0-9])*x$".r
+  val allEs8xBelowEs811x = "^es8(1[1-9]|[2-9][0-9]|[1-9][0-9]{2,})x$".r
```
Comment (mateuszkp96) [inline tests-utils/src/main/scala/tech/beshu/ror/utils/misc/EsModule.scala:67]: Let's use `^es8(0?[0-9]|10)x$`

File: es811x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/EsqlIndicesEsRequestContext.scala [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import tech.beshu.ror.es.utils.EsqlRequestHelper
+import tech.beshu.ror.exceptions.SecurityPermissionException
+import tech.beshu.ror.implicits.*
+import tech.beshu.ror.syntax.*
+
+class EsqlIndicesEsRequestContext private(actionRequest: ActionRequest with CompositeIndicesRequest,
+                                         esContext: EsContext,
+                                         aclContext: AccessControlStaticContext,
+                                         clusterService: RorClusterService,
+                                         override val threadPool: ThreadPool)
```
Comment (mateuszkp96) [inline es811x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/EsqlIndicesEsRequestContext.scala:46]: Formatting (Missing space)

File: es811x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/EsqlIndicesEsRequestContext.scala [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  extends BaseFilterableEsRequestContext[ActionRequest with CompositeIndicesRequest](actionRequest, esContext, aclContext, clusterService, threadPool) {
+
+  override protected def requestFieldsUsage: RequestFieldsUsage = RequestFieldsUsage.NotUsingFields
+
+  private lazy val requestClassification = EsqlRequestHelper.classifyEsqlRequest(actionRequest) match {
+    case result@Right(_) => result
+    case result@Left(ClassificationError.ParsingException) => result
+    case Left(ClassificationError.UnexpectedException(ex)) =>
+      throw RequestSeemsToBeInvalid[CompositeIndicesRequest](s"Cannot extract SQL indices from ${actionRequest.getClass.show}", ex)
```
Comment (mateuszkp96) [inline es811x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/EsqlIndicesEsRequestContext.scala:55]: Maybe:  cannot extract ESQL..? Here and in other places

File: es811x/src/main/scala/tech/beshu/ror/es/utils/EsqlRequestHelper.scala [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    ReflecUtils
+      .getMethodOf(request.getClass, Modifier.PUBLIC, "query", 1)
+      .invoke(request, newQuery)
+    request
+  }
+
+  private def newQueryFrom(oldQuery: String, requestTables: NonEmptyList[IndexTable], finalIndices: Set[String]) = {
+    requestTables.toList.foldLeft(oldQuery) {
+      case (currentQuery, table) =>
+        val (beforeFrom, afterFrom) = currentQuery.splitBy("FROM")
```
Comment (mateuszkp96) [inline es811x/src/main/scala/tech/beshu/ror/es/utils/EsqlRequestHelper.scala:112]: There is a tip in docs: `ES|QL keywords are case-insensitive`, so we should be aware of both cases.
It seems that SQL is affected too

**Review by mateuszkp96 — CHANGES_REQUESTED** [review]

File: tests-utils/src/main/scala/tech/beshu/ror/utils/misc/EsModule.scala (hunk 1 above)
Comment (coutoPL) [reply tests-utils/src/main/scala/tech/beshu/ror/utils/misc/EsModule.scala]: or maybe `^es8([0-9]|10)x$`?

**Review by mateuszkp96 — APPROVED** [review]
LGTM

**Pull request merged**